File: Java/libraries/recognizers-text/src/main/java/com/microsoft/recognizers/text/matcher/MatchResult.java
package com.microsoft.recognizers.text.matcher;

import java.util.HashSet;
import java.util.Set;

public class MatchResult<T> {
    private int start;
    private int length;
    private T text;
    private Set<String> canonicalValues;

    public MatchResult(int start, int lenght, HashSet<String> canonicalValues) {
        this.start = start;
        this.length = lenght;
        this.canonicalValues = canonicalValues;
    }

    public MatchResult(int start, int length) {
        this(start, length, new HashSet<>());
    }

    public MatchResult() {
        this(0, 0);
    }

    public int getStart() {
        return start;
    }

    public void setStart(int start) {
        this.start = start;
    }

    public int getLength() {
        return length;
    }

    public void setLength(int length) {
        this.length = length;
    }

    public T getText() { return text; }

    public void setText(T text) {
        this.text = text;
    }

    public Set<String> getCanonicalValues() {
        return canonicalValues;
    }

    public void setCanonicalValues(Set<String> canonicalValues) {
        this.canonicalValues = canonicalValues;
    }

    public  int getEnd() { return getStart()+getLength(); }
}

File: Python/libraries/recognizers-text/recognizers_text/Matcher/match_result.py
class MatchResult:

    def __init__(self, start: int = 0, length: int = 0, ids: [] = []):
        self.__length = length,
        self.__start = start,
        self.__canonical_values = ids,
        self.__text = ''

    @property
    def text(self) -> str:
        return self.__text

    @text.setter
    def text(self, text) -> str:
        self.__text = text

    @property
    def start(self) -> int:
        return self.__start

    @start.setter
    def start(self, start) -> int:
        self.__start = start

    @property
    def length(self) -> int:
        return self.__length

    @length.setter
    def length(self, length) -> int:
        self.__length = length

    @property
    def end(self) -> int:
        return self.start + self.length

    @property
    def text(self) -> []:
        return self.__text

    @text.setter
    def text(self, text) -> []:
        self.__text = text

    @property
    def canonical_values(self) -> []:
        return self.__canonical_values

    @canonical_values.setter
    def canonical_values(self, canonical_values) -> []:
        self.__canonical_values = canonical_values
